src/arrayMethodSort.js

```diff
@@ -4,8 +4,38 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const initialCompareFunction = (a, b) => {
+      if (String(a) < String(b)) {
+        return -1;
+      }
+
+      return 1;
+    };
+
+    const compare = compareFunction ?? initialCompareFunction;
+
+    let isSwapped = false;
+
+    do {
+      isSwapped = false;
+
+      for (let idx = 0; idx < this.length; idx++) {
+        const result = compare(this[idx], this[idx + 1]);
```

Comment:
The comparison function should be called with two arguments. When `idx` is the last index, `this[idx + 1]` will be `undefined`, which may not be handled correctly by the comparison function. Ensure the comparison function is only called with valid elements of the array.